examples/llava/llava.cpp

```diff
@@ -109,28 +174,49 @@ int main(int argc, char ** argv) {
         llama_backend_free();
         free(image_embd);
 
-        return 1;
+        return NULL;
+    }
+
+    {
+        printf("\n%s: image encoded in %8.2f ms by CLIP (%8.2f ms per image patch)\n", __func__, t_img_enc_ms, t_img_enc_ms / n_img_pos);
     }
 
-    // process the prompt
-    // llava chat format is "<system_prompt>USER: <image_embeddings>\n<textual_prompt>\nASSISTANT:"
 
+    auto ctx_llava = (struct llava_context *)malloc(sizeof(llava_context));
+
+    ctx_llava->ctx_llama = ctx_llama;
+    ctx_llava->ctx_clip = ctx_clip;
+    ctx_llava->model = model;
+    ctx_llava->image_embd = image_embd;
+    ctx_llava->n_img_pos = n_img_pos;
+    return ctx_llava;
+
+}
+
+void llava_free(struct llava_context * ctx_llava) {
+    llama_free(ctx_llava->ctx_llama);
+    llama_free_model(ctx_llava->model);
+    llama_backend_free();
+    free(ctx_llava->image_embd);
+}
+
+void llava_process_prompt(struct llava_context * ctx_llava, gpt_params * params, const char * prompt) {
```

Comment:
Let's avoid such god-like functions for now --it kills hackability in early stages of development --it brings no benefit over functions in `llava-utils.h`. Better to have single-responsibility functions as much as possible to enhance the development speed and flexibility.

Reply:
👍 sure